x-pack/legacy/plugins/maps/public/layers/styles/vector/components/color/color_stops.js

```diff
@@ -6,66 +6,49 @@
 
 import _ from 'lodash';
 import React from 'react';
-import PropTypes from 'prop-types';
-
-import {
-  EuiColorPicker,
-  EuiFormRow,
-  EuiFieldNumber,
-  EuiFlexGroup,
-  EuiFlexItem,
-  EuiButtonIcon,
-} from '@elastic/eui';
-import { addRow, removeRow, isColorInvalid, isStopInvalid, isInvalid } from './color_stops_utils';
-
-const DEFAULT_COLOR = '#FF0000';
-
-export const ColorStops = ({ colorStops = [{ stop: 0, color: DEFAULT_COLOR }], onChange }) => {
+import { removeRow, getColorInput, getDeleteButton, getColorStopRow } from './color_stops_utils';
+
+export const ColorStops = ({
+  onChange,
+  colorStops,
+  isStopsInvalid,
+  sanitizeStopInput,
+  getStopError,
+  renderStopInput,
+  addNewRow,
+  canDeleteStop,
+}) => {
   function getStopInput(stop, index) {
     const onStopChange = e => {
       const newColorStops = _.cloneDeep(colorStops);
-      const sanitizedValue = parseFloat(e.target.value);
-      newColorStops[index].stop = isNaN(sanitizedValue) ? '' : sanitizedValue;
+      const newValue = sanitizeStopInput(e.target.value);
+      newColorStops[index].stop = newValue;
```

Comment:
You could prob just compress these two lines since `newValue` isn't used anywhere else:
`newColorStops[index].stop = sanitizeStopInput(e.target.value);`